blocks/model.py

```diff
@@ -1,75 +1,105 @@
-"""Defines models.
+"""Model - heavily annotated computation graph.
 
-A model is a thin layer of abstraction between the user-defined computation
-graph, bricks, parameters and main loop extensions. This module provides
-the basic :class:`AbstractModel` interface as well as its implementations
-(currently only :class:`Model`).
+A model in Blocks is simply an annotated computation graph.  The class
+:class:`Model` extends :class:`blocks.graph.ComputationGraph` :class:,
+which is able to handle annotations and roles in general, but is
+deliberately made unaware of specific annotations that a Theano graph
+created by Blocks typically has, such as bricks and application calls.  The
+:class:`Model` adds this functionality. Using :class:`Model` you can do
+things like quering all the bricks used to build the computation graph,
+requesting "hierarhical names" of the parameters, etc.
```

Comment:
request*, to match the tense of query above; hierarchical*, also, not really clear from context what this means